uaclient/api/u/pro/subscription/v1.py

```diff
@@ -0,0 +1,265 @@
+            "created_at", DatetimeDataValue, doc="Creation date of the account"
+        ),
+        Field(
+            "external_account_ids",
+            data_list(StringDataValue),
+            doc="External account identifiers",
+        ),
+        Field("id", StringDataValue, doc="Account id"),
+        Field("name", StringDataValue, doc="Account name"),
+    ]
+
+    def __init__(
+        self,
+        *,
+        created_at: str,
+        external_account_ids: List[str],
+        id: str,
+        name: str,
+    ):
+        self.created_at = created_at
+        self.external_account_ids = external_account_ids
+        self.id = id
+        self.name = name
+
+
+class SubscriptionResult(DataObject, AdditionalInfo):
+    fields = [
+        Field("contract", ContractInfo, doc="Contract Information"),
+        Field("account", AccountInfo, doc="Account information"),
+        Field("machine_id", StringDataValue, doc="Machine id"),
+        Field("activity_id", StringDataValue, doc="Activity id"),
+    ]
+
+    def __init__(
+        self,
+        *,
+        contract: ContractInfo,
+        account: AccountInfo,
+        machine_id: str,
+        activity_id: str,
+    ):
+        self.contract = contract
+        self.account = account
+        self.machine_id = machine_id
+        self.activity_id = activity_id
+
+
+def subscription() -> SubscriptionResult:
+    return _subscription(UAConfig())
+
+
+def _subscription(cfg: UAConfig) -> SubscriptionResult:
+    """
+    Returns the Ubuntu Pro subscription information for the machine.
+    """
+    if not _is_attached(cfg).is_attached:
+        return SubscriptionResult(
+            contract=ContractInfo(
+                created_at="",
+                id="",
+                name="",
+                products=[],
+                tech_support_level="",
+                origin=None,
+                effective=None,
+                expires=None,
+            ),
+            account=AccountInfo(
+                created_at="",
+                external_account_ids=[],
+                id="",
+                name="",
+            ),
+            machine_id="",
+            activity_id="",
+        )
+
+    machine_token_file = machine_token.get_machine_token_file(cfg)
+    machineTokenInfo = machine_token_file.machine_token["machineTokenInfo"]
+
+    # Contract Info
+    contractInfo = machineTokenInfo["contractInfo"]
+    tech_support_level = UserFacingStatus.INAPPLICABLE.value
+    expires = (
+        machine_token_file.contract_expiry_datetime
+        if contractInfo.get("effectiveTo")
+        else contractInfo.get("effectiveTo", None)
+    )
+    effective = contractInfo.get("effectiveFrom", None)
+
+    support = (
+        machine_token_file.entitlements().get("support", {}).get("entitlement")
+    )
+    if support:
+        supportLevel = support.get("affordances", {}).get("supportLevel")
+        if supportLevel:
+            tech_support_level = supportLevel
```

Comment:
maybe it is better to create a property directly on `machine_token_file` for this. My rationale is that all logic need to extract information from the machine token file should live in that class